trajopt_optimizers/trajopt_sqp/test/cart_position_optimization_unit.cpp

```diff
@@ -0,0 +1,181 @@
+    tesseract_environment::AdjacencyMap::Ptr adjacency_map = std::make_shared<tesseract_environment::AdjacencyMap>(
+        tesseract->getEnvironment()->getSceneGraph(),
+        forward_kinematics->getActiveLinkNames(),
+        tesseract->getEnvironment()->getCurrentState()->link_transforms);
+    auto kinematic_info = std::make_shared<trajopt::CartPosKinematicInfo>(
+        forward_kinematics, adjacency_map, Eigen::Isometry3d::Identity(), forward_kinematics->getTipLinkName());
+
+    // Get target position
+    Eigen::VectorXd start_pos(forward_kinematics->numJoints());
+    start_pos << 0.0, 0, 0, -1.0, 0, -1, -0.00;
+    if (DEBUG)
+      std::cout << "Joint Limits:\n" << forward_kinematics->getLimits().transpose() << std::endl;
+
+    auto target_pose = Eigen::Isometry3d::Identity();
+    joint_target_ = start_pos;
+    forward_kinematics->calcFwdKin(target_pose, joint_target_);
+    //    auto world_to_base =
+    //        tesseract->getEnvironment()->getCurrentState()->link_transforms.at(forward_kinematics->getBaseLinkName());
+    //    target_pose = world_to_base * target_pose;
+
+    // 3) Add Variables
+    std::vector<trajopt::JointPosition::Ptr> vars;
+    for (int ind = 0; ind < 1; ind++)
+    {
+      auto zero = Eigen::VectorXd::Zero(7);
+      auto var = std::make_shared<trajopt::JointPosition>(
+          zero, forward_kinematics->getJointNames(), "Joint_Position_" + std::to_string(ind));
+      vars.push_back(var);
+      nlp_.AddVariableSet(var);
+    }
+
+    // 4) Add constraints
+    for (const auto& var : vars)
+    {
+      auto cnt = std::make_shared<trajopt::CartPosConstraint>(target_pose, kinematic_info, var);
+      nlp_.AddConstraintSet(cnt);
+      //    cnt->LinkWithVariables(nlp_.GetOptVariables());
+      //    auto cost = std::make_shared<trajopt::SquaredCost>(cnt);
+      //    nlp_.AddCostSet(cost);
+    }
+
+    if (DEBUG)
+    {
+      nlp_.PrintCurrent();
+      std::cout << "Constraint Jacobian: \n" << nlp_.GetJacobianOfConstraints() << std::endl;
+    }
+  }
+};
+
+/**
+ * @brief Applies a cartesian position constraint and solves the problem with IPOPT
+ */
+// TEST_F(CartPositionOptimization, cart_position_optimization_ipopt)
+//{
+//  ifopt::Problem nlp_ipopt(nlp_);
+//  ifopt::IpoptSolver solver;
+//  solver.SetOption("derivative_test", "first-order");
+//  solver.SetOption("linear_solver", "mumps");
+//  //  ipopt.SetOption("jacobian_approximation", "finite-difference-values");
+//  solver.SetOption("jacobian_approximation", "exact");
+//  solver.SetOption("print_level", 5);
+
+//  // solve
+//  solver.Solve(nlp_ipopt);
+//  Eigen::VectorXd x = nlp_ipopt.GetOptVariables()->GetValues();
+
+//  for (Eigen::Index i = 0; i < joint_target_.size(); i++)
+//    EXPECT_NEAR(x[i], joint_target_[i], 1e-5);
+
+//  if (DEBUG)
+//  {
+//    std::cout << x.transpose() << std::endl;
+//    nlp_ipopt.PrintCurrent();
+//  }
+//}
```

Comment:
Should this be removed or have a TODO?

Reply:
I can remove this. Ultimately I think we need to move these tests into a test suite and then have them all run against all of the solvers that we have.